testsuite/smoke-test/src/txn_broadcast.rs

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::smoke_test_environment::SwarmBuilder;
+use crate::test_utils::{assert_balance, create_and_fund_account, transfer_coins};
+use aptos_config::config::NodeConfig;
+use forge::{NodeExt, Swarm, SwarmExt};
+use std::sync::Arc;
+use std::time::{Duration, Instant};
+
+/// Checks txn goes through consensus even if the local validator is not creating proposals.
+/// This behavior should be true with both mempool and quorum store.
+#[tokio::test]
+async fn test_txn_broadcast() {
+    let mut swarm = SwarmBuilder::new_local(4)
+        .with_aptos()
+        .with_init_config(Arc::new(|_, conf, _| {
+            conf.api.failpoints_enabled = true;
+        }))
+        .build()
+        .await;
+    let transaction_factory = swarm.chain_info().transaction_factory();
+    let version = swarm.versions().max().unwrap();
+    let validator_peer_ids = swarm.validators().map(|v| v.peer_id()).collect::<Vec<_>>();
+
+    let validator = validator_peer_ids[1];
+    let vfn = swarm
+        .add_validator_fullnode(
+            &version,
+            NodeConfig::default_for_validator_full_node(),
+            validator,
+        )
+        .unwrap();
+
+    for validator in swarm.validators_mut() {
```

Comment:
doesn't building of the swarm already wait for validators being healthy?

Reply:
Good catch! This is no longer needed now that validators don't restart after attaching a vfn.